src/block/request.rs

```diff
@@ -181,6 +189,7 @@ impl Request {
     }
 
     /// Parses a request from a given `desc_chain`.
+    /// TODO split this gigantic function in smaller pieces.
```

Comment:
:+1:

Reply:
Hmm looks like the `Discard` and `WriteZeroes` support is noticeably more complex than the rest. Does it make sense to have two PRs? We can focus first on the previous commits and then create another PR for the rest.